src/transports.ts

```diff
@@ -36,10 +40,16 @@ export class TransportKeeper {
     }
   }
 
-  public register(
-    typeName: string,
-    handler: TransportHandler
-  ) {
+  public subscribe(typeName: string, onMessage: TransportMessageHandler) {
```

Comment:
Perhaps for the contracts unification we should use the common first argument type like [here](https://github.com/jolocom/jolocom-sdk/pull/127/files#diff-d9f84c9d0970a2d8db8df4a10410818764d9d2041695ec7ad3736fadf8f7554bR61)? (e.g. `TransportDesc`)

Reply:
I'm not exactly sure what the `config` from `TransportDesc` is supposed to include, but what's usually put there is the `callbackURL` for an interaction. Doesn't seem like `register` and `subscribe` need the config so far.

Reply:
The `'config'` option is optional so the call from the client could be like:
`transportKeeper.subscribe({ type: 'http' }, ...)`
and the method api like:
`public subscribe(transport: TransportDesc, onMessage: TransportMessageHandler) { ... }`

But it i's okay to leave as it is so please skip this comment...

Reply:
It's just weird that we would subscribe with a transport description without a config, and inside it we would start the transport with a different transport description (which now contains a config, specifically the `callbackURL`).